.pipelines/templates/aks-engine-setup.yaml

```diff
@@ -21,7 +21,8 @@ steps:
     condition: and(succeeded(), eq(variables['CLUSTER_TYPE'], 'aks-engine'))
 
   - script: |
-      envsubst < test/e2e/cluster_configs/$(CLUSTER_CONFIG).json > cluster.json
+      OS_TYPE=$(echo $(CLUSTER_CONFIG) | cut -d '-' -f1 | tr -d '[:space:]')
```

Comment:
`OS_TYPE` is computed multiple times across different templates. Could we instead do it once and export it in the main template `e2e-test`?